pandas/io/sql.py

```diff
@@ -956,8 +962,42 @@ def __init__(
         if not len(self.name):
             raise ValueError("Empty table name specified")
 
+    def _drop_temporary_table(self):
+        """Drop a temporary table. Temporary tables are not in a database's meta data
+        and need to be dropped hard coded."""
+        if self.schema is None:
+            query = f"DROP TABLE {self.name}"
+        else:
+            query = f"DROP TABLE {self.schema}.{self.name}"
+        self.pd_sql.execute(query)
+
+    def _exists_temporary(self):
+        """Check if a temporary table exists. Temporary tables are not in a database's
+        meta data. The existence is duck tested by a SELECT statement."""
+        from sqlalchemy.exc import (
+            OperationalError,
+            ProgrammingError,
+        )
+
+        if self.schema is None:
+            query = f"SELECT * FROM {self.name} LIMIT 1"
+        else:
+            query = f"SELECT * FROM {self.schema}.{self.name} LIMIT 1"
+        try:
+            _ = self.pd_sql.read_query(query)
+            return True
+        except ProgrammingError:
```

Comment:
Does sqlalchemy not provide a higher level abstraction than ProgrammingError or OperationalError? I'm not sure I understand the distinction between those, and I am not sure if they would catch all possible errors thrown by sqlalchemy either

Reply:
The is a higher abstraction possible with `DatabaseError`.